### PR TITLE
feat: get all community detail from given slug

### PR DESCRIPTION
**Pull request by spantheslayer**
## What does this PR do?
Implements feature from #3332 

## Related issues
NA

Fixes # (issue)
NA

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Enhancement (non-breaking small changes to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Explanation of the changes
Added api to search for all the communitites from given slug which will be used to implement searching for communities in search bar.

### CODE REVIEW

**Comment by spantheslayer** [comment]
hey @bigint  could you please check this PR out?

File: packages/workers/communities/src/handlers/get/getCommunitySlugs.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  try {
+    const client = new Client(env.DB_URL);
+    await client.connect();
+
+    const query = {
+      text: `
+      SELECT *
+      FROM communities
+      WHERE slug LIKE '%' || $1 || '%'; 
+      `,
```
Comment (bigint) [inline packages/workers/communities/src/handlers/get/getCommunitySlugs.ts:20]:
```suggestion
      text: `
        SELECT *
        FROM communities
        WHERE slug LIKE '%' || $1 || '%'; 
      `,
```

File: packages/workers/communities/src/handlers/get/getCommunitySlugs.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  try {
+    const client = new Client(env.DB_URL);
+    await client.connect();
+
+    const query = {
+      text: `
+      SELECT *
+      FROM communities
+      WHERE slug LIKE '%' || $1 || '%';
```
Comment (bigint) [inline packages/workers/communities/src/handlers/get/getCommunitySlugs.ts:19]: Can we also scan for name, is this using postgres full text search?

File: packages/workers/communities/src/handlers/get/getCommunitySlugs.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      FROM communities
+      WHERE slug LIKE '%' || $1 || '%'; 
+      `,
+      values: [slug]
+    };
+
+    const result = await client.query(query);
+    const communutites = result.rows;
+
+    return new Response(JSON.stringify(communutites));
```
Comment (bigint) [inline packages/workers/communities/src/handlers/get/getCommunitySlugs.ts:27]:
```suggestion
    const result = await client.query(query);

    return new Response(JSON.stringify(result.rows));
```

File: packages/workers/communities/src/index.ts [hunk 4]
```diff
@@ -53,6 +54,9 @@ router.get('/getMembers/:communityId/:offset', ({ params }, env) =>
 router.get('/getIsMember/:communityId/:profileId', ({ params }, env) =>
   getIsMember(params.communityId, params.profileId, env)
 );
+router.get('/getCommunitySlugs/:slug', ({ params }, env) =>
```
Comment (bigint) [inline packages/workers/communities/src/index.ts:57]: Rename to /search

Also `:slug` should be a query and not path just for standards, replacing URL is costly than updating the query

Eg: /search?query=lenster

**Review by bigint — CHANGES_REQUESTED** [review]

**Pull request closed without merging**